pkgs/build-support/setup-hooks/make-binary-wrapper/default.nix

```
@@ -0,0 +1,29 @@
+{ stdenv
+, lib
+, darwin
+, makeSetupHook
+, dieHook
+, writeShellScript
+, tests
+, cc ? stdenv.cc
+, sanitizers ? []
+}:
+
+makeSetupHook {
+  deps = [ dieHook ]
+    # https://github.com/NixOS/nixpkgs/issues/148189
+    ++ lib.optional (stdenv.isDarwin && stdenv.isAarch64) darwin.cctools;
+
+  substitutions = {
+    cc = let
+      san = lib.escapeShellArgs (map (s: "-fsanitize=${s}") sanitizers);
+    in "${cc}/bin/cc ${san}";
```

```suggestion
    cc = "${cc}/bin/cc ${lib.escapeShellArgs (map (s: "-fsanitize=${s}") sanitizers)}";
```
Why do we even need other sanitize flags for aarch64?